### PR TITLE
after_party task to fix transition_aged_youth when DOB is nil

### LINKED ISSUE #1996: Court Report Transitioning Case Detection
### Impacted User Types
volunteers

### Environment
local, QA

### Current Behavior
court reports generate non transitioning elements in the court report template for transitioning cases(gold text)

### Expected Behavior
court reports **do not** generate non transitioning elements in the court report template for transitioning cases(black text only)

### How to Replicate
ex: 
1. - Log into QA as a volunteer.
2. - Click on "Generate Reports" in the left sidebar menu.
3. - Select the transitioning case.
4. - Click "Generate report" at the bottom of the page.

Possibly caused by casa cases without birthdates

### PR DESCRIPTION
**Pull request by librod89**
### What github issue is this PR for, if any?
Resolves #1996

### What changed, and why?

- Add an after_party task to fix an invalid scenario in staging data
- Fix seeder so this scenario doesn't happen again

### How will this affect user permissions?
It does not affect user permissions

### CODE REVIEW

File: spec/controllers/emancipations_controller_spec.rb [hunk 1]
```diff
@@ -14,12 +14,12 @@
```
Comment (librod89) [inline spec/controllers/emancipations_controller_spec.rb:14]: This and the following spec were changed when I ran `bundle exec standardrb --fix`

File: lib/tasks/deployment/20210507200644_fix_transition_aged_youth_for_nil_birth_month_year_youth.rake [hunk 2]
```diff
@@ -0,0 +1,16 @@
+namespace :after_party do
+  desc "Deployment task: fix_transition_aged_youth_for_nil_birth_month_year_youth"
+  task fix_transition_aged_youth_for_nil_birth_month_year_youth: :environment do
+    puts "Running deploy task 'fix_transition_aged_youth_for_nil_birth_month_year_youth'"
```
Comment (librod89) [inline lib/tasks/deployment/20210507200644_fix_transition_aged_youth_for_nil_birth_month_year_youth.rake:4]: Do after_party tasks run on prod? Should I add an `unless Rails.env.production?`

Comment (compwron) [reply lib/tasks/deployment/20210507200644_fix_transition_aged_youth_for_nil_birth_month_year_youth.rake:4]: Yes, they do. Please do.

File: lib/tasks/deployment/20210507200644_fix_transition_aged_youth_for_nil_birth_month_year_youth.rake [hunk 3]
```diff
@@ -0,0 +1,17 @@
+namespace :after_party do
```
Comment (compwron) [inline lib/tasks/deployment/20210507200644_fix_transition_aged_youth_for_nil_birth_month_year_youth.rake:1]: I would like to have tests for this also but let's merge for now :)

**Review by compwron — APPROVED** [review]

**Pull request merged**